.github/workflows/go.yml

```diff
@@ -0,0 +1,133 @@
+name: Go Code
+
+on:
+  push:
+    paths:
+    - '*.go'
+    - '.revive.toml'
+    - 'cmd/**'
+    - 'go.mod'
+    - 'go.sum'
+    - 'Makefile'
+    - 'pkg/**'
+    - '!pkg/webui/**'
+    - 'pkg/webui/**.go'
+    - 'tools/**'
+    - '!tools/mage/docs.go'
+    - '!tools/mage/js.go'
+    - '!tools/mage/proto.go'
+    - '!tools/mage/scripts/fix-grpc-gateway-names.sh'
+    - '!tools/mage/styl.go'
+    - '!tools/mage/translations.js'
+    - '!tools/mage/version.go'
+    - '!tools/mage/xx.js'
+
+  pull_request:
+    paths:
+    - '*.go'
+    - '.revive.toml'
+    - 'cmd/**'
+    - 'go.mod'
+    - 'go.sum'
+    - 'Makefile'
+    - 'pkg/**'
+    - '!pkg/webui/**'
+    - 'pkg/webui/**.go'
+    - 'tools/**'
+    - '!tools/mage/docs.go'
+    - '!tools/mage/js.go'
+    - '!tools/mage/proto.go'
+    - '!tools/mage/scripts/fix-grpc-gateway-names.sh'
+    - '!tools/mage/styl.go'
+    - '!tools/mage/translations.js'
+    - '!tools/mage/version.go'
+    - '!tools/mage/xx.js'
+
+jobs:
+  quality:
+    name: Code Quality
+    runs-on: ubuntu-18.04
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+    - name: Initialize Go cache
+      uses: actions/cache@v2
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Make Mage
+      run: make tools/bin/mage
+    - name: Remove unnecessary type conversions
+      run: tools/bin/mage go:unconvert
+    - name: Format code
+      run: tools/bin/mage go:fmt
+    - name: Lint code
+      run: tools/bin/mage go:lint
+    - name: Generate i18n messages
+      run: tools/bin/mage go:messages
+    - name: Generate allowed fieldmask paths
+      run: tools/bin/mage jsSDK:allowedFieldMaskPaths
+    - name: Check for diff
+      run: tools/bin/mage git:diff
+
+  test:
+    name: Tests
+    runs-on: ubuntu-18.04
+    services:
+      postgres:
+        image: postgres
+        ports:
+        - '5432/tcp'
+        env:
+          POSTGRES_USER: root
+          POSTGRES_PASSWORD: root
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis
+        ports:
+          - '6379/tcp'
+    steps:
+    - name: Create ttn_lorawan_is_test DB
+      uses: docker://postgres
+      env:
+        PGPASSWORD: root
+      with:
+        entrypoint: /usr/bin/createdb
+        args: -h postgres -U root ttn_lorawan_is_test
+    - name: Create ttn_lorawan_is_store_test DB
+      uses: docker://postgres
+      env:
+        PGPASSWORD: root
+      with:
+        entrypoint: /usr/bin/createdb
+        args: -h postgres -U root ttn_lorawan_is_store_test
+    - name: Set up Go
+      uses: actions/setup-go@v2
```

Comment:
Let's use

```yaml
with:
  go-version: '~1.14'
```

Reply:
Is there a reason not to just use the latest?

Reply:
Because

1. We had problems with using the "latest" version of things in the past.
1. We want to be able to *choose* to use a new Go version, and not be forced to.
1. We want to be able to re-run a build from the past and still get (almost) the same result.